src/cr/cube/cube_slice.py

```diff
@@ -25,134 +47,104 @@ def _update_args(self, kwargs):
             # pass them to the underlying cube (which is the slice).
             return kwargs
 
+        # Handling API methods that include 'axis' parameter
+
         axis = kwargs.get('axis')
-        if axis is None or isinstance(axis, tuple):
-            # If no axis was passed, we don't need to update anything. If axis
-            # was a tuple, it means that the caller was expecting a 3D cube.
-            return kwargs
+        # Expected usage of the 'axis' parameter from CubeSlice is 0, 1, or
+        # None. CrunchCube handles all other logic. The only 'smart' thing
+        # about the handling here, is that the axes are increased for 3D cubes.
+        # This way the 3Dness is hidden from the user and he still sees 2D
+        # crosstabs, with col and row axes (0 and 1), which are transformed to
+        # corresponding numbers in case of 3D cubes (namely 1 and 2). In the
+        # case of None, we need to analyze across all valid dimensions, and the
+        # CrunchCube takes care of that (no need to update axis if it's None).
```

Comment:
Can axis be a tuple?

Reply:
Yes it can, but it shouldn't. I've updated the comment rather than explicitly raising. We can though, if you feel that that's better (I'm indecisive).